controller/watch/dynamic_resource_client.go

```diff
@@ -0,0 +1,22 @@
+package watch
+
+import (
+	"k8s.io/client-go/1.5/dynamic"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/pkg/watch"
+)
+
+type dynamicResourceClient interface {
+	Watch(*v1.ListOptions) (watch.Interface, error)
+}
+
+type realDynamicResourceClient dynamic.ResourceClient
```

Comment:
@martinmaly this is simply an implementation of the `dynamicResourceClient` interface (above) that uses the `dynamic.ResourceClient`.

I had to change the `thirdPartyWatcher` `func` in `watcher.go` to take an impl. of that interface so I could more easily unit-test that function, so I've created a fake version (similar to the `FakeWatcher` in the k8s client) and this real version.